src/monmove.cpp

```diff
@@ -271,6 +277,113 @@ float monster::rate_target( Creature &c, float best, bool smart ) const
     return INT_MAX;
 }
 
+std::map<item *, const tripoint> monster::find_loot_in_radius( const tripoint &target, int radius )
+{
+    bool steals_food = has_flag( MF_STEALS_FOOD );
+    bool steals_shiny = has_flag( MF_STEALS_SHINY );
+    bool suitable = false;
+    std::map<item *, const tripoint> lootable;
+
+    //Check all tiles nearby within radius
+    for( const tripoint &p : g->m.points_in_radius( target, radius ) ) {
+        //Check if we can see some items on the tile
+        if( g->m.sees_some_items( p, *this ) ) {
+            //A candidate tile with some items!
+            auto items = g->m.i_at( p );
+            //Check all items on tile
+            for( item &itm : items ) {
+                suitable = false;
+
+                //Does monster try to steal food and is it actually food?
+                if( steals_food && ( itm.is_comestible() && itm.get_comestible()->comesttype == "FOOD" ) ) {
+                    if( !itm.rotten() ) {
+                        suitable = true;
+                    }
+
+                    //TODO: Sanity checks here on what we're trying to loot.
+                }
+
+                //Does monster steal shiny items?
+                if( steals_shiny &&
+                    ( itm.made_of( material_id( "gold" ) ) ||
+                      itm.made_of( material_id( "silver" ) ) ||
+                      itm.made_of( material_id( "diamond" ) ) ||
+                      itm.made_of( material_id( "platinum" ) ) ) ) {
+
+                    //TODO: Sanity checks on what we're trying to loot here
+
+                    suitable = true;
+                }
+
+
+                if( suitable ) {
+                    lootable.insert( std::make_pair( &itm, p ) );
+                }
+            }
+        }
+    }
+
+    return lootable;
+}
+
+bool monster::eat_from_inventory()
+{
+    auto it = inv.begin();
+    for( item &itm : inv ) {
+        //Is food?
+        if( ( itm.is_comestible() && itm.get_comestible()->comesttype == "FOOD" ) ) {
+            if( itm.charges > 1 ) {
+                itm.mod_charges( -1 );
+            }
+
+            if( g->u.sees( *this ) ) {
+                add_msg( m_warning, "%1$s eats some of %2$s!", name().c_str(), itm.display_name() );
+            }
+
+            //Remove item
+            inv.erase( it );
+            mod_moves( -50 );
+            return true;
+        }
+
+        it++;
+    }
+
+
+    return false;
+}
+
+item_location monster::select_desired_loot( std::map<item *, const tripoint> &loot )
+{
+    std::map<item *, const tripoint>::iterator it;
+    item *desired_i = nullptr;
+    tripoint desired_p;
+    int desired_range = MONSTER_LOOT_DIST + 1;
+
+    item *i;
+    tripoint p;
+    int d;
```

Comment:
All these variables are used only in the cycle below, so they should be placed inside it.